proxy/core/src/main/scala/io/cloudstate/proxy/HttpApi.scala

```diff
@@ -517,35 +523,49 @@ object HttpApi {
         )
 
       if (methDesc.isServerStreaming) {
-        def toChunk(entityMessage: MessageOrBuilder): HttpEntity.Chunk =
-          HttpEntity.Chunk(
-            if (isHttpBodyResponse) extractDataFromHttpBody(entityMessage) ++ NEWLINE_BYTES
-            else ByteString(jsonPrinter.print(entityMessage)) ++ NEWLINE_BYTES
-          )
+        val sseAccepted = new MediaTypeNegotiator(grpcRequest.headers).isAccepted(MediaTypes.`text/event-stream`)
 
         data
           .prefixAndTail(1)
           .runWith(Sink.head)
-          .map {
+          .flatMap {
             case (Seq(protobuf: ProtobufAny), rest) =>
-              val entityMessage: MessageOrBuilder = parseResponseBody(protobuf)
-              val contentType =
-                if (isHttpBodyResponse) extractContentTypeFromHttpBody(entityMessage)
-                else ContentTypes.`application/json`
-
-              val first =
-                Source.single(toChunk(entityMessage))
-              val following =
-                rest.map(protobuf => toChunk(parseResponseBody(protobuf)))
-
-              //val extensions = extractExtensionsFromHttpBody(entityMessage) // FIXME / TODO HttpBody.extensions not supported yet
-
-              HttpResponse(
-                entity = HttpEntity.Chunked(
-                  contentType,
-                  first.concat(following)
+              if (isHttpBodyResponse) {
+                val entityMessage = parseResponseBody(protobuf) // We need to peek-ahead to check the content-type of the response we're sending out
+                Future.successful(
```

Comment:
@raboof Would it make sense to create a ToResponseMarshaller for this instead?

Reply:
I'm not sure if it's worth it. I wonder if we could provide/generalize some concepts at the akka-grpc side to make this easier to write here.